RefApp/Utilities/TableViewDataSource.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  TableViewDataSource.swift
+//  RefApp
+//
+//  Created by Udaya Sri Senarathne on 2018-11-21.
+//  Copyright © 2018 emp. All rights reserved.
+//
+
+import UIKit
+import Exposure
+
+
+/// Generic TableViewDataSource Class
+class TableViewDataSource<Model>: NSObject, UITableViewDataSource {
+    
+    typealias CellConfigurator = (Model, UITableViewCell) -> Void
+    var models: [Model] = []
+    
+    private let reuseIdentifier: String
+    private let cellConfigurator: CellConfigurator
+    
+    init(models: [Model],
+         reuseIdentifier: String,
+         cellConfigurator: @escaping CellConfigurator) {
+        self.models = models
+        self.reuseIdentifier = reuseIdentifier
+        self.cellConfigurator = cellConfigurator
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if models.count == 0 {
+            tableView.showEmptyMessage(message: NSLocalizedString("Sorry no data to show", comment: ""))
+        } else {
+            tableView.hideEmptyMessage()
+        }
+        return models.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let model = models[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        cellConfigurator(model, cell)
```

Comment:
Nothing major in this context but what is the entire purpose of `cellConfugurator(...)`?

Just setting cell appearance from already known data?

If so, I suggest using 
`func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath)` to customise the appearance of the cell as it arrives.

That would of cause complicate things by making `TableViewDataSource` both the tableview's datasource and delegate